processor/schemaprocessor/config.go

```diff
@@ -0,0 +1,76 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemaprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/schema"
+)
+
+var (
+	errInvalidSchemaURL = errors.New("invalid URL schema")
+)
+
+// Config defines the user provided values for the Schema Processor
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"`
+
+	// SchemaURL defines a url where the schema can be loaded from
+	// and then used to then evaluate translations between the configured version
+	SchemaURL string `mapstructure:"schema_url"`
```

Comment:
I still don't understand the purpose of this.

The input data is going to contain a SchemaURL on its own. What happens if the SchemaURL in the input is different from the one specified here? Are we trying to match this value with the one in the input? Is this for  preloading the schema file? What if the input schema is different than what is specified here?

Instead, why are we not simply not using the SchemaURL in the input data, load that schema file on demand (we can cache it too, even persistently) and do our work using that schema file?

Reply:
Wait, I think I may have misunderstood this from the start, I presumed the schema processor had no idea on the location of where the translations files were stored and were part of the configuration so that organisations or vendors could also use this in the event that they require an extension of what exist in the existing semantic convention.

Sorry, I should of laid out my assumptions earlier.